test/pthread/test_pthread_proxying.c

```diff
@@ -314,6 +324,186 @@ void test_proxy_callback_with_ctx(void) {
   destroy_widget(&w13);
 }
 
+em_promise_result_t explode(void** result, void* data, void* value) {
+  printf("error!");
+  abort();
+  return EM_PROMISE_REJECT;
+}
+
+em_promise_result_t test_promise_self(void** result, void* data, void* value) {
+  widget* w14 = data;
+
+  em_promise_t promise =
+    emscripten_proxy_promise(proxy_queue, pthread_self(), do_run_widget, w14);
+
+  *result = emscripten_promise_then(promise, check_widget, explode, w14);
```

Comment:
I wonder if a NULL third argument here should default to explode/abort?

Reply:
No, that would get in the way of normal rejection propagation. Right now it's possible to chain up a bunch of promises and only supply a single rejection handler at the end. If any of the intermediate promises reject, that final rejection handler will be invoked, just like it would with JS promises. I should probably add a test for that.